docs/gadgets/index.mdx

```diff
@@ -6,11 +6,330 @@ description: Documentation about the official Gadgets"
+```bash
+# built-in
+$ sudo ig trace open -o columns=comm,path
+COMM             PATH
+...
+cat              /lib/libm.so.6
+cat              /lib/libresolv.so.2
+cat              /lib/libc.so.6
+cat              /tmp/foo
+
+# image-based
+$ sudo ig run trace_open --fields proc.comm,fname
+COMM             FNAME
+...
+cat              /lib/libm.so.6
+cat              /lib/libresolv.so.2
+cat              /lib/libc.so.6
+cat              /tmp/foo
+```
+
+#### Pulling Gadget Images
+
+Image-based Gadgets aren't shipped with the `ig` binary, they are instead stored
+as OCI artifacts on container registries. It means they need to be pulled from
+the internet when running them for the first time. If you're running in an
+environment where it's not possible to reach the internet while running the
+gadgets, you have the following options:
+
+##### Use your own container registry
+
+Mirror the Gadget images to your own container registry.
+
+```bash
+$ sudo ig image pull trace_open:v0.36.0
+$ sudo ig image tag trace_open:v0.36.0 mycontainer_registry/trace_open:v0.36.0
+$ sudo ig image push mycontainer_registry/trace_open:v0.36.0
+```
+
+Then, run the image from a machine with access to your container registry
+
+```bash
+$ sudo ig run mycontainer_registry/trace_open:v0.36.0
+```
+
+##### Use the image export and import functionality
+
+Download a local copy of the images on a machine with internet access:
+
+```bash
+$ sudo ig image pull trace_open:v0.36.0
+$ sudo ig image pull trace_exec:v0.36.0
+# ...
+
+# export the images to a file
+$ sudo ig image export trace_open:v0.36.0 trace_exec:v0.36.0 > myimages.tar
+```
+
+Then, transfer `myimages.tar` to the machine where you want to run the Gadgets,
+import and run them:
+
+```bash
+$ sudo ig image import myimages.tar
+$ sudo ig run trace_open:v0.36.0
+```
+
+#### New Features
+
+This is a raw list of features that are only available with the image-based
+gadgets. Please check the individual documentation for each one of them to get
+more information.
+
+- [**Headless mode**](../reference/headless): Run Gadgets on background.
+- [**Gadget Instance Manifests**](../reference/manifests): Run a Gadget from a
+  file containing its configuration.
+- [**Metrics**](../reference/export-metrics): Export metrics to Prometheus and
+  OpenTelemetry.
+- **Uniform filtering**: Filtering by common fields like command name, pid, tid,
+  etc. on the kernel directly.
+- [**Creating your own Gadgets**](../gadget-devel/): You can now create your own
+  Gadgets without requiring any changes on the project.
+
+### Golang API Users
+
+The main difference for the users of the Golang API is that image-based Gadgets
+don't provide a Golang package, instead a generic runner needs to be used to run
+them. We suggest to check the
+[examples](https://github.com/inspektor-gadget/inspektor-gadget/tree/%IG_BRANCH%/examples/gadgets)
+available to get a better understanding of how to run image-based Gadgets
+directly from your application.
```

Comment:
Add a second paragraph to say that all options are preserved when using the Golang API:
1. Download from internet
2. Load from the local oci store on the filesystem
2. Load from a tar file
2. Embed in your own binary with `go:embed`

Reply:
I updated it, however I think we need to improve https://www.inspektor-gadget.io/docs/latest/api/golang. Probably by including all the examples there with some more explanation.

Reply:
This will be handled in #3948.